02-peer-protocol.md

```diff
@@ -580,31 +596,50 @@ The receiving node:
   - if `fee_satoshis` is equal to its previously sent `fee_satoshis`:
     - SHOULD sign and broadcast the final closing transaction.
     - MAY close the connection.
-  - otherwise, if `fee_satoshis` is greater than
-the base fee of the final commitment transaction as calculated in
-[BOLT #3](03-transactions.md#fee-calculation):
-    - MUST fail the connection.
-  - if `fee_satoshis` is not strictly
-between its last-sent `fee_satoshis` and its previously-received
-`fee_satoshis`, UNLESS it has since reconnected:
+  - if `fee_satoshis` matches its previously sent `fee_range`:
+    - SHOULD use `fee_satoshis` to sign and broadcast the final closing transaction
+    - SHOULD reply with a `closing_signed` with the same `fee_satoshis` value if it is different from its previously sent `fee_satoshis`
+    - MAY close the connection.
+  - if the message contains a `fee_range`:
+    - if there is no overlap between that and its own `fee_range`:
+      - SHOULD send a `warning`
```

Comment:
Wait, why a `warning`? At this point you're unable to agree and are force-closing, thus it should be an `error`.

Reply:
I find it too strict and unnecessary to force-close in that case, especially since that `fee_range` will usually be provided manually by a node operator that can adapt it depending on the response from its peer. A warning is the best mechanism we currently have for that (disconnecting doesn't make much sense because at reconnection you must send the exact same `closing_signed` which will trigger another disconnection, which doesn't give the node operator the opportunity to send a second `closing_signed` with a different set of `fee_range`).

Reply:
There's a few ways we could go with this, but I don't think simply "send a warning, but leave the channel in an unusable state" with no further guidance is an acceptable one (especially since the "manual intervention" process isn't defined here).

For example, we could say that the node that initiated the shutdown MUST either send an error or send a new closing_signed with a new fee range (and any updates needed to ensure starting over with a new fee range is allowed), plus eg that the node that did not initiate the shutdown MUST send a warning and fail the channel after N minutes if a new closing_signed isn't sent.

Reply:
You're totally right, I went with a half-measure here, we can surely improve that.

I don't think responding with a different `fee_range` that doesn't overlap with the sender's `fee_range` helps though, it would either create a loop where each side sends a different `fee_range` or we'd be reintroducing something very similar to the previous fee negotiation mechanism (which we want to get rid of!).

What do you think about this variation of your last proposal:

- if you receive a `fee_range` that you disagree with, you send a warning, and force-close after N minutes if your peer doesn't send you another `closing_signed` with a different `fee_range`
- if you sent a `closing_signed` and didn't receive a response after N minutes, force-close

NB: this N minutes counter should probably take into account whether the peer is offline or not (you may want to only count N minutes when he is online), but that's really an implementation policy, not something that belongs in the spec

Reply:
> I don't think responding with a different fee_range that doesn't overlap with the sender's fee_range helps though, it would either create a loop where each side sends a different fee_range or we'd be reintroducing something very similar to the previous fee negotiation mechanism (which we want to get rid of!).

I don't see why not? As long as the new fee range is only sent in response to user input and not in an automated way, we still have the nice "here's my fee range, if you accept, pick something in the fee range and sign and then we're done" property.

> What do you think about this variation of your last proposal:

I thought that's what I was proposing, but if I was unclear, yep, what you wrote sounds good!

Reply:
Done in https://github.com/lightningnetwork/lightning-rfc/pull/847/commits/034486cb49d8c50fbad2c845109684a0048aeafb, let me know if that's better.